include/xtensor/xlayout.hpp

```diff
@@ -0,0 +1,22 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille and Sylvain Corlay                     *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XLAYOUT_HPP
+#define XLAYOUT_HPP
+
+namespace xt
+{
+    enum class layout
+    {
+        row_major,
+        column_major,
+        dynamic
+    };
+}
+
+#endif
```

Comment:
I think it isn't possible to forward declare an enum, but I needed the declaration in `forward_xtensor`. We could also put the enum in `forward_xtensor.hpp`. What do you think?